### ISSUE TITLE
Idea - enable multiple backends via Substrait

### ISSUE THREAD

**Issue by jonrmayer** [uwdata/mosaic#375]
Probably a fair bit of work (!!) - but it would be awesome to abstract away the mosaic sql/query engine and thus leverage the Substrait ecosystem.

See:  https://substrait.io/

Especially: https://substrait.io/community/powered_by/

**Comment by jheer** [comment]
Agreed! A related idea is to use DuckDB itself, leveraging [multi-DB support](https://duckdb.org/2024/01/26/multi-database-support-in-duckdb.html).

p.s. moving this issue over to Discussions!
